lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -326,17 +341,39 @@ public T ReceiveFromChildren(IReduceFunction<T> reduceFunction)
             var receivedData = new List<T>();
             var childrenToReceiveFrom = new HashSet<string>(_children.Select(node => node.Identifier));
 
+            //Logger.Log(Level.Info, "*************entering receive frm children***************");
+            
             while (childrenToReceiveFrom.Count > 0)
             {
-                NodeStruct childWithData = GetNodeWithData();
-                byte[][] data = ReceiveFromNode(childWithData, false);
-                if (data == null || data.Length != 1)
+                var childrenWithData = GetNodeWithData(childrenToReceiveFrom);
+
+                //Logger.Log(Level.Info, "+++++++++++received data from +++++++++++++ " + childrenWithData.ToList().Count);
+
+                /*foreach (var child in childrenWithData)
                 {
-                    throw new InvalidOperationException("Received invalid data from child with id: " + childWithData.Identifier);
+                    Logger.Log(Level.Info, "++++++++++Children Id: " + child.Identifier);
+                }*/
+
+               // + childWithData.Identifier);
+                foreach (var child in childrenWithData)
+                {
+                    byte[][] data = ReceiveFromNode(child);
```

Comment:
Original code was not to remove the data from the queue. Why it was like that and why make the change?